Kernel/Jail.h

```diff
@@ -37,12 +38,21 @@ class Jail : public RefCounted<Jail> {
     SpinlockProtected<size_t, LockRank::None>& attach_count() { return m_attach_count; }
 
 private:
-    Jail(NonnullOwnPtr<KString>, JailIndex);
+    Jail(NonnullOwnPtr<KString>, JailIndex, ScopedProcessList&);
```

Comment:
Might be nicer to pass a NNRP here and move it? Would make it clearer that the jail will keep the process list alive, which is not obvious from the plain reference.